docs/functions-suggestion-handlers.md

```diff
@@ -0,0 +1,218 @@
+select drop down menu. Here's an example function (which we will assume exists in
+a `mod.ts` file under the `functions/quote/` subdirectory in your app) that posts
+a message with a external data select drop down menu:
+
+```typescript
+import { SlackFunction } from "deno-slack-sdk/mod.ts";
+// QuoteFunction is the function we defined in the previous section
+import { QuoteFunction } from "./definition.ts";
+
+export default SlackFunction(QuoteFunction, async ({ inputs, client }) => {
+  console.log('Incoming quote request!');
+
+  await client.chat.postMessage({
+    channel: inputs.channel_id,
+    blocks: [{
+      "type": "actions",
+      "block_id": "so-inspired",
+      "elements": [{
+        type: "external_select",
+        placeholder: {
+          type: "plain_text",
+          text: "Inspire",
+        },
+        action_id: "ext_select_input",
+      },
+      {
+        type: "button",
+        text: {
+          type: "plain_text",
+          text: "Post",
+        },
+        action_id: "post_quote",
+      }],
+    }],
+  });
+  // Important to set completed: false! We should set the function's complete
+  // status later - in the action handler responding to the button click
+  return {
+    completed: false,
+  };
+});
+```
+
+The key bit of information we need to remember before moving on to adding a
+suggestion handler are the `action_id` and `block_id` properties defined in the `blocks`
+payload. Using these IDs, we will be able to differentiate between the different
+Block Kit components that users interacted with in this message.
+
+### Adding Block Suggestion Handlers
+
+The [Deno Slack SDK][sdk] - which comes bundled in your generated Run-on-Slack
+application - provides a means for defining a handler to execute every time a user
+interacts with an interactive Block Kit element created by your function.
+
+Continuing with our above example, we can now define a handler that will listen
+for interactions with the external data drop down menu. The code to add a Block Kit
+suggestion handler is "chained" off of your top-level function, and would look like this:
+
+```typescript
+export default SlackFunction(QuoteFunction, async ({ inputs, client }) => {
+  // ... the rest of your QuoteFunction logic here ...
+}).addBlockSuggestionHandler(
+  "ext_select_input", // The first argument to addBlockActionsHandler can accept an action_id string, among many other formats!
+  // Check the API reference at the end of this document for the full list of supported options
+  async ({ body, client }) => { // The second argument is the handler function itself
+    console.log('Incoming suggestion handler invocation', body);
+    // Fetch some inspirational quotes
+    const apiResp = await fetch("https://motivational-quote-api.herokuapp.com/quotes");
+    const quotes = await apiResp.json();
+    console.log('Returning', quotes.length, 'quotes');
+    const opts = {
```

Comment:
I forgot the exact number but we may want to mention the maximum number of options/option_groups here for more developer-friendliness

Reply:
I couldn't find a reference to what is the maximum number of options one can return... I did end up posting in our api docs channel about this issue: https://slack-pde.slack.com/archives/C8DPMVAQY/p1666275726696799